File: src/app/api/models/operation-dto.ts
/* tslint:disable */
/* eslint-disable */
import { Category } from '../models/category';
import { ThirdParty } from '../models/third-party';
import { Vault } from '../models/vault';
export interface OperationDto {
  amount?: number;
  category?: Category;
  causal?: string;
  endDate?: string;
  id?: number;
  isProgrammed?: boolean;
  startDate?: string;
  thirdPartys?: ThirdParty;
  type?: 'REVENUE' | 'EXPENSE';
  vault?: Vault;
}

File: src/main/java/osiride/vitt_be/utils/OperationDTO.java
package osiride.vitt_be.utils;

import java.math.BigDecimal;
import java.time.LocalDate;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import osiride.vitt_be.domain.Category;
import osiride.vitt_be.domain.ThirdParty;
import osiride.vitt_be.domain.Vault;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class OperationDTO {

	private Long id;
	private String causal;
	private BigDecimal amount;
	private Boolean isProgrammed;
	private LocalDate startDate;
	private LocalDate endDate;
	private Vault vault;
	private Category category;
	private ThirdParty thirdParty;
	private TransactionType type;
}
